strawberry/ext/mypy_plugin.py

```diff
@@ -245,17 +253,114 @@ def enum_hook(ctx: DynamicClassDefContext) -> None:
     )
 
 
-def strawberry_pydantic_class_callback(ctx: ClassDefContext):
+def add_method_to_class(
+    api: SemanticAnalyzerPluginInterface,
+    cls: ClassDef,
+    name: str,
+    args: List[Argument],
+    return_type: Type,
+    self_type: Optional[Type] = None,
+    tvar_def: Optional[TypeVarDef] = None,
+    is_static: bool = False,
+) -> None:
+    """Adds a new method to a class definition.
+    Edited to add is_static for
+    https://github.com/python/mypy/blob/master/mypy/plugins/common.py
+    because couldn't figure out how to add static methods otherwise"""
+    info = cls.info
```

Comment:
ok abit dumb but i just couldn't figure out how to add a static method. otherwise would keep complaining about wanting self passed in. anyone has an idea?

Reply:
also, i'm pretty sure this breaks for people on older mypy versions.

Reply:
looking at the original code it seems that it hard codes self as an argument :)

sounds like we could a PR to mypy to support creating static methods :)

Reply:
yep. i'll submit it one day. i'll add it to my mental backlog. although mypy's 170 open PRs do scare me....

Reply:
I managed to get one PR in, I'm sure they'll accept yours too 😊 

Or maybe you can open an issue (or I can do it too if you're busy) to begin with

Reply:
i'll do it. :D